### PR TITLE
[fix](Tvf) return empty set when tvf queries an empty file or an error uri

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number:

### Before:
return errors when tvf queries an empty file or an error uri:
1. get parsed schema failed, empty csv file
2. Can not get first file, please check uri.

### Now:
we just return empty set when tvf queries an empty file or an error uri.
```sql
mysql> select * from s3( 
"uri" = "https://error_uri/exp_1.csv", 
"s3.access_key"= "xx", 
"s3.secret_key" = "yy", 
"format" = "csv") limit 10;

Empty set (1.29 sec)
```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by BePPPower** [comment]
run buildall

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 1]
```diff
@@ -527,6 +527,9 @@ Status CsvReader::get_parsed_schema(std::vector<std::string>* col_names,
     bool is_parse_name = false;
     RETURN_IF_ERROR(_prepare_parse(&read_line, &is_parse_name));
 
+    if (_file_reader->size() == 0) {
```
Comment (morningman) [inline be/src/vec/exec/format/csv/csv_reader.cpp:530]: Need to add comment to explain why return OK when file size is 0

File: be/src/vec/exec/format/csv/csv_reader.cpp [hunk 2]
```diff
@@ -924,7 +923,7 @@ Status CsvReader::_parse_col_names(std::vector<std::string>* col_names) {
     // no use of _line_reader_eof
     RETURN_IF_ERROR(_line_reader->read_line(&ptr, &size, &_line_reader_eof, _io_ctx));
     if (size == 0) {
-        return Status::InternalError("The first line is empty, can not parse column names");
+        return Status::OK();
```
Comment (morningman) [inline be/src/vec/exec/format/csv/csv_reader.cpp:926]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java [hunk 3]
```diff
@@ -344,6 +344,10 @@ public List<Column> getTableColumns() throws AnalysisException {
         TNetworkAddress address = new TNetworkAddress(be.getHost(), be.getBrpcPort());
         try {
             PFetchTableSchemaRequest request = getFetchTableStructureRequest();
+            if (request == null) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/ExternalFileTableValuedFunction.java:347]: unify it to `fillColumns`

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run feut

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/LocalTableValuedFunction.java [hunk 4]
```diff
@@ -86,9 +86,9 @@ private void getFileListFromBackend() throws AnalysisException {
             Future<PGlobResponse> response = proxy.glob(address, requestBuilder.build());
             PGlobResponse globResponse = response.get(5, TimeUnit.SECONDS);
             if (globResponse.getStatus().getStatusCode() != 0) {
-                throw new AnalysisException(
-                        "error code: " + globResponse.getStatus().getStatusCode()
-                                + ", error msg: " + globResponse.getStatus().getErrorMsgsList());
+                LOG.warn("get file from backend failed. error code: {}, error msg: {}",
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/tablefunction/LocalTableValuedFunction.java:89]: I think we should keep this exception.
For local tvf, it mains used for debug of POC, better expose every exception

**Review by morningman — DISMISSED** [review]
LGTM

**Comment by morningman** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/LocalTableValuedFunction.java (hunk 4 above)
Comment (BePPPower) [reply fe/fe-core/src/main/java/org/apache/doris/tablefunction/LocalTableValuedFunction.java]: done

**Comment by BePPPower** [comment]
run p0

**Comment by BePPPower** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
lgtm

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**